missing `return` on error in `fallsWithinCidrValidator.ValidateString()` leads to nil pointer Derek

These crashes happen when the `fallsWithinCidrValidator` parses invalid host and/or subnet notation.

The validator detects the problem and registers a diagnostic event with TF core, but does not exit.

```text
Stack trace from the terraform-provider-apstra_v0.20.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12c55a0]

goroutine 29 [running]:
net.networkNumberAndMask(0xc000396320?)
        net/ip.go:498
net.(*IPNet).Contains(0xc000396320?, {0x0, 0x0, 0x1?})
        net/ip.go:522 +0x27
terraform-provider-apstra/apstra/apstra_validator.fallsWithinCidrValidator.ValidateString({{0x0, {0xc0004e4560, 0x2, 0x2}}, 0x0, 0x0}, {0x1b2ad90, 0xc0004f4bd0}, {{{0xc000599b90, 0x1, ...}}, ...}, ...)
        terraform-provider-apstra/apstra/apstra_validator/falls_within_cidr.go:99 +0x879
```

--- a/apstra/apstra_validator/falls_within_cidr.go
+++ b/apstra/apstra_validator/falls_within_cidr.go
@@ -88,11 +88,13 @@ func (o fallsWithinCidrValidator) ValidateString(ctx context.Context, req valida
 						mp, fmt.Sprintf("error parsing CIDR block %q",
 							mpString.ValueString()), err.Error(),
 					)
+					return
 				}
 			} else {
 				resp.Diagnostics.Append(validatordiag.BugInProviderDiagnostic(
 					fmt.Sprintf("attribute at %q must be types.String got %t",
 						o.expression, mpVal)))
+				return
 			}
 
 			ip := net.ParseIP(req.ConfigValue.ValueString())
@@ -101,6 +103,7 @@ func (o fallsWithinCidrValidator) ValidateString(ctx context.Context, req valida
 					req.Path,
 					fmt.Sprintf("value must fall within %s", subnet.String()),
 					req.ConfigValue.ValueString()))
+				return
 			}
 
 			if ip.Equal(allZeros) && !o.allZerosOk {
@@ -108,6 +111,7 @@ func (o fallsWithinCidrValidator) ValidateString(ctx context.Context, req valida
 					req.Path,
 					"value must not be the all-zeros address %s",
 					req.ConfigValue.ValueString()))
+				return
 			}
 
 			allOnes := netaddr.BroadcastAddr(subnet)
@@ -116,6 +120,7 @@ func (o fallsWithinCidrValidator) ValidateString(ctx context.Context, req valida
 					req.Path,
 					"value must not be the all-ones address %s",
 					req.ConfigValue.ValueString()))
+				return
 			}
 		}
 	}

